jlm/tooling/CommandLine.cpp

```diff
@@ -98,6 +98,90 @@ JlmOptCommandLineOptions::Reset() noexcept
   Optimizations_.clear();
 }
 
+JlmOptCommandLineOptions::OptimizationId
+JlmOptCommandLineOptions::FromCommandLineArgument(const std::string& commandLineArgument)
+{
+  static std::unordered_map<std::string, OptimizationId> map(
+    {
+      {AaSteensgaardAgnosticCommandLineArgument_,     OptimizationId::AASteensgaardAgnostic},
+      {AaSteensgaardRegionAwareCommandLineArgument_,  OptimizationId::AASteensgaardRegionAware},
+      {CommonNodeEliminationCommandLineArgument_,     OptimizationId::cne},
+      {DeadNodeEliminationCommandLineArgument_,       OptimizationId::dne},
+      {FunctionInliningCommandLineArgument_,          OptimizationId::iln},
+      {InvariantValueRedirectionCommandLineArgument_, OptimizationId::InvariantValueRedirection},
+      {NodePushOutCommandLineArgument_,               OptimizationId::psh},
+      {NodePullInCommandLineArgument_,                OptimizationId::pll},
+      {NodeReductionCommandLineArgument_,             OptimizationId::red},
+      {ThetaGammaInversionCommandLineArgument_,       OptimizationId::ivt},
+      {LoopUnrollingCommandLineArgument_,             OptimizationId::url}
+    });
+
+  if (map.find(commandLineArgument) != map.end())
+    return map[commandLineArgument];
+
+  throw util::error("Unknown command line argument: " + commandLineArgument);
+}
+
+const char*
+JlmOptCommandLineOptions::ToCommandLineArgument(OptimizationId optimizationId)
+{
+  static std::unordered_map<OptimizationId, const char*> map(
```

Comment:
This is simply the reverse of the above unordered_map.
Isn't it possible to define one and then invert key-value to automatically create the other one, so that there's no need to make sure that the two lists are consistent?